thirdparty/github.go

```diff
@@ -653,23 +833,52 @@ func GithubAuthenticate(ctx context.Context, code, clientId, clientSecret string
 }
 
 // GetTaggedCommitFromGithub gets the commit SHA for the given tag name.
-func GetTaggedCommitFromGithub(ctx context.Context, oauthToken, owner, repo, tag string) (string, error) {
-	client := getGithubClientRetry(oauthToken, "GetTaggedCommitFromGithub")
-	defer utility.PutHTTPClient(client)
+func GetTaggedCommitFromGithub(ctx context.Context, token, owner, repo, tag string) (string, error) {
+	installationToken, _ := GetInstallationToken(ctx, owner, repo, nil)
 
-	url := fmt.Sprintf("https://api.github.com/repos/%s/%s/git/refs/tags/%s", owner, repo, tag)
-	resp, err := client.Get(url)
+	httpClient := getGithubClientRetry(installationToken, "GetTaggedCommitFromGithub")
+	defer utility.PutHTTPClient(httpClient)
+	client := github.NewClient(httpClient)
+
+	tags, resp, err := client.Repositories.ListTags(ctx, owner, repo, nil)
 	if resp != nil {
 		defer resp.Body.Close()
+		if err == nil {
```

Comment:
nit: seems like you could avoid the heavily nested blocks (heavily nested block are generally not preferred) by just doing the for loop after the `if resp != nil` block on `tags`. The err in this case seems kind of irrelevant since you're searching for a specific tag result, and if it errored, then `tags` will be nil anyways and the for loop will be skipped.